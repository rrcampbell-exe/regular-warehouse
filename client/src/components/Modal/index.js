import React, { useRef, useEffect, useState } from 'react';

const Modal = () => {

  const regularVideo = document.getElementById("regular-video");

  const modalEl = useRef(null)

  const [isModalOpen, setIsModalOpen] = useState(false);

  // opens modal after visitor has been on page for X amount of time
  useEffect(() => {
    setTimeout(() => {
      modalEl.current.classList.add("modal-container-visible");
      modalEl.current.classList.remove("modal-container-invisible")
      setIsModalOpen(isModalOpen)
    }, 30000)
    return () => {}
  });

  function removeVideo() {
    regularVideo.classList.remove("regular-video-visible");
    regularVideo.classList.add("regular-video-invisible");
    document.exitFullscreen();
  }

  function closeModal() {
    setIsModalOpen(!isModalOpen)
    modalEl.current.classList.add("modal-container-invisible");
    modalEl.current.classList.remove("modal-container-visible");
    
  }
  
  function playVideo() {
    regularVideo.classList.remove("regular-video-invisible");
    regularVideo.classList.add("regular-video-visible");
    regularVideo.requestFullscreen()
    regularVideo.play()
    setTimeout(removeVideo, 65000)
    closeModal()
  }

  return (
    <div ref={modalEl} className="modal-container-invisible container" id="modal">
      
      <div className="col-6 modal-text">
        <h3 className="text-center">Is the coast clear? ðŸ‘€</h3>
        <div className="d-flex flex-row justify-content-around">
          <button className='btn btn-outline-secondary' onClick={playVideo}><span className="material-icons align-bottom">
check
</span></button>
          <button className='btn btn-outline-secondary' onClick={closeModal}><span className="material-icons align-bottom">
block
</span></button>
        </div>
        <span className="text-center">If <span className="material-icons align-bottom">
check
</span>, there may be some flashing lights.</span>
      </div>
      <div className="col-6 cat-container">
        <img alt="A grayish brown cat peeks around a corner" className="secret-cat" src="./assets/creeping-cat.png"/>
      </div>
    </div>
  )
}

export default Modal;