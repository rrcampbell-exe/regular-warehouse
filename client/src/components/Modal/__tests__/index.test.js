import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Modal from '..';

afterEach(cleanup);

describe('Modal component', () => {
    it('renders', () => { // this test passes
        render(<Modal/>)
    })
})