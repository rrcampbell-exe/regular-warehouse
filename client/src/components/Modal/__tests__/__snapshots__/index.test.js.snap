// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="modal-container-invisible container"
    id="modal"
  >
    <div
      class="col-6 cat-container"
    >
      <img
        class="secret-cat"
        src="./assets/creeping-cat.png"
      />
    </div>
    <div
      class="col-6 modal-text"
    >
      <h3>
        Is the coast clear? 👀
      </h3>
      <div
        class="button-container"
      >
        <button>
          🕺🐈💃
        </button>
        <button>
          😓😬😨
        </button>
      </div>
      <span>
        If 🕺🐈💃, there may be some flashing lights.
      </span>
    </div>
  </div>
</DocumentFragment>
`;
